TaskModules/powershell/Azure/SPNCreation.ps1

```diff
@@ -10,9 +10,102 @@ param
     [string] $spnRole = "owner",
     
     [Parameter(Mandatory=$false, HelpMessage="Provide Azure environment name for your subscription")]
-    [string] $environmentName = "AzureCloud"
+    [string] $environmentName = "AzureCloud",
+
+    [Parameter(Mandatory=$false, HelpMessage="Provide AzureStackManagementURL to add AzureStack environment to AzureRmEnvironments.")]
+    [string] $azureStackManagementURL
 )
 
+$AZURESTACK_ENVIRONMENT = "AzureStack"
+
+function Add-AzureStackToAzureRmEnvironment {
+    param (
+        [Parameter(mandatory=$true, HelpMessage="The Admin ARM endpoint URI of the Azure Stack Environment")]
+        $EndpointURI,
+        [parameter(mandatory=$true, HelpMessage="Azure Stack environment name for use with AzureRM commandlets")]
+        [string] $Name
+    )
+
+    $EndpointURI = $EndpointURI.TrimEnd("/")
+
+    $Domain = ""
+    try {
+        $uriendpoint = [System.Uri] $EndpointURI
+        $i = $EndpointURI.IndexOf('.')

```

Comment:
How safe it is to assume that anything after first dot is domain?
http://www.w3.org/Addressing/URL/url-spec.txt

Reply:
This URL is always in format of management.<domain>,  Same method Azure Stack team use for provisioning and our tasks use the same algorithm and has already been verified.